src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -154,46 +154,100 @@ type job = Available_job.t [@@deriving sexp]
 (*Scan state and any zkapp updates that were applied to the to the most recent
    snarked ledger but are from the tree just before the tree corresponding to
    the snarked ledger*)
-[%%versioned
+module Repr = struct
+  [%%versioned
+  module Stable = struct
+    module V2 = struct
+      type t =
+        { scan_state :
+            ( Ledger_proof_with_sok_message.Stable.V2.t
+            , Transaction_with_witness.Stable.V2.t )
+            Parallel_scan.State.Stable.V1.t
+        ; previous_incomplete_zkapp_updates :
+            Transaction_with_witness.Stable.V2.t list
+            * [ `Border_block_continued_in_the_next_tree of bool ]
+        }
+      [@@deriving sexp]
+
+      let to_latest = Fn.id
+
+      let hash (t : t) =
+        let state_hash =
+          Parallel_scan.State.hash t.scan_state
+            (Binable.to_string (module Ledger_proof_with_sok_message.Stable.V2))
+            (Binable.to_string (module Transaction_with_witness.Stable.V2))
+        in
+        let ( previous_incomplete_zkapp_updates
+            , `Border_block_continued_in_the_next_tree continue_in_next_tree ) =
+          t.previous_incomplete_zkapp_updates
+        in
+        let incomplete_updates =
```

Comment:
Why implementation was changed from the following?

They seem equivalent, but I'm not sure yet

```ocaml
let incomplete_updates =
        List.fold ~init:(Digestif.SHA256.init ())
          previous_incomplete_zkapp_updates ~f:(fun h t ->
            Digestif.SHA256.feed_string h
            @@ Binable.to_string (module Transaction_with_witness.Stable.V2) t )
        |> Digestif.SHA256.get
```